src/transformers/models/imagebind/convert_imagebind_to_hf.py

```diff
@@ -0,0 +1,221 @@
+                ]
+            )
+    # fmt: on
+
+    return rename_keys
+
+
+# here we list all keys to be renamed (original name on the left, our name on the right)
+def create_rename_keys(config):
+    vision_config = config.vision_config
+    text_config = config.text_config
+    audio_config = config.audio_config
+
+    rename_keys = []
+
+    # fmt: off
+
+    # Convert Vision
+    rename_keys.extend([
+        ("modality_preprocessors.vision.cls_token", "vision_model.embeddings.cls_token"),
+        ("modality_preprocessors.vision.rgbt_stem.proj.1.weight", "vision_model.embeddings.patch_embedding.projection.weight"),
+        ("modality_preprocessors.vision.pos_embedding_helper.pos_embed", "vision_model.embeddings.position_embeddings"),
+        ("modality_heads.vision.0.weight", "vision_model.layernorm.weight"),
+        ("modality_heads.vision.0.bias", "vision_model.layernorm.bias"),
+        ("modality_heads.vision.2.weight", "vision_projection.weight"),
+        ("modality_trunks.vision.pre_transformer_layer.0.weight", "vision_model.pre_layernorm.weight"),
+        ("modality_trunks.vision.pre_transformer_layer.0.bias", "vision_model.pre_layernorm.bias"),
+    ])
+
+    rename_keys.extend(
+        rename_encoder_layers(vision_config, "vision")
+    )
+
+    # Convert Text
+    rename_keys.extend([
+        ("modality_preprocessors.text.pos_embed", "text_model.embeddings.position_embedding.weight"),
+        ("modality_preprocessors.text.token_embedding.weight", "text_model.embeddings.token_embedding.weight"),
+        ("modality_heads.text.proj.0.weight", "text_model.layernorm.weight"),
+        ("modality_heads.text.proj.0.bias", "text_model.layernorm.bias"),
+        ("modality_heads.text.proj.1.weight", "text_projection.weight"),
+        ("modality_postprocessors.text.1.log_logit_scale", "text_postprocessor.log_logit_scale"),
+    ])
+
+    rename_keys.extend(
+        rename_encoder_layers(text_config, "text")
+    )
+
+    # Convert Audio
+    rename_keys.extend([
+        ("modality_preprocessors.audio.cls_token", "audio_model.embeddings.cls_token"),
+        ("modality_preprocessors.audio.rgbt_stem.proj.weight", "audio_model.embeddings.patch_embedding.projection.weight"),
+        ("modality_preprocessors.audio.rgbt_stem.norm_layer.weight", "audio_model.embeddings.patch_embedding.layernorm.weight"),
+        ("modality_preprocessors.audio.rgbt_stem.norm_layer.bias", "audio_model.embeddings.patch_embedding.layernorm.bias"),
+        ("modality_preprocessors.audio.pos_embedding_helper.pos_embed", "audio_model.embeddings.position_embeddings"),
+        ("modality_heads.audio.0.weight", "audio_model.layernorm.weight"),
+        ("modality_heads.audio.0.bias", "audio_model.layernorm.bias"),
+        ("modality_heads.audio.2.weight", "audio_projection.weight"),
+    ])
+
+    rename_keys.extend(
+        rename_encoder_layers(audio_config, "audio")
+    )
+    # fmt: on
+
+    return rename_keys
```

Comment:
I don't want to be a pain 😅 but a lot of this can be simplified with regexed! Would love to see something simple like we have in `mllama`! https://github.com/huggingface/transformers/blob/144852fb6bbe584e9ff7d13511180aec42e1b366/src/transformers/models/mllama/convert_mllama_weights_to_hf.py#L41

Reply:
Made the changes while using regex as `re` in the recent commits :+1: . Please check and mention if any changes are necessary.